packages/amplify-util-uibuilder/src/clients/amplify-studio-client.ts

```diff
@@ -56,19 +69,56 @@ export default class AmplifyStudioClient {
       resolvedEnvName,
       resolvedAppId,
     ])) as ServiceConfigurationOptions;
-    return new AmplifyStudioClient(awsConfigInfo, resolvedAppId, resolvedEnvName);
+
+    const client = new AmplifyStudioClient(awsConfigInfo, resolvedAppId, resolvedEnvName);
+
+    if ((await getTransformerVersion(context)) === 2) {
+      await client.loadMetadata();
+      client.isSupportedGraphQL = true;
+    } else {
+      client.isSupportedGraphQL = false;
+    }
+
+    return client;
   }
 
   /*
-  * Builds the clients
-  */
+   * Builds the clients
+   */
   constructor(awsConfigInfo: ServiceConfigurationOptions, appId: string, envName: string) {
     this.#amplifyUiBuilder = buildAmplifyUiBuilderClient(awsConfigInfo);
     this.#amplifyBackend = buildAmplifyBackendClient(awsConfigInfo);
     this.#appId = appId;
     this.#envName = envName;
+    this.metadata = {
+      autoGenerateForms: false,
+      autoGenerateViews: false,
+    };
   }
 
+  /**
+   * loads metadata from appId/envName
+   */
+  loadMetadata = async (envName?: string, appId?: string): Promise<void> => {
+    const environmentName = envName || this.#envName;
+    const resolvedAppId = appId || this.#appId;
+    try {
+      const response = await this.#amplifyUiBuilder
+        .getMetadata({
+          appId: resolvedAppId,
+          environmentName,
+        })
+        .promise();
+      this.metadata = {
+        autoGenerateForms: response.features?.autoGenerateForms === 'true',
+        autoGenerateViews: response.features?.autoGenerateViews === 'true',
+      };
+    } catch (e) {
+      printer.debug(e.toString());
```

Comment:
Removing the printer statements in error since we debug errors at the top level.
We do however want the debug statements on the entities coming in from the sdk, as it helps debug the codegen process.